pkg/machinery/client/grpc_connection_wrapper.go

```diff
@@ -0,0 +1,50 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+package client
+
+import (
+	"context"
+
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/metadata"
+)
+
+type GRPCConnectionWrapper struct {
```

Comment:
we can embed the connection into a struct

```go
type grpcConnectionWrapper struct {
   *grpc.ClientConn
   
    clusterName string
}
```

this way all methods of `ClientConn` will be exposed via the wrapper by default, but we can override the methods we need (`Invoke` & `NewStream`). This way we can drop `Close`

also this struct shouldn't be public, as we don't need it in the  external API